otelcol/collector.go

```diff
@@ -246,11 +264,36 @@ func (col *Collector) DryRun(ctx context.Context) error {
 	return col.validatePipelineCfg(ctx, cfg, factories)
 }
 
+func newFallbackLogger(options []zap.Option) (*zap.Logger, error) {
+	ec := zap.NewProductionEncoderConfig()
+	ec.EncodeTime = zapcore.ISO8601TimeEncoder
+	zapCfg := &zap.Config{
+		Level:            zap.NewAtomicLevelAt(zapcore.DebugLevel),
+		Encoding:         "console",
+		EncoderConfig:    ec,
+		OutputPaths:      []string{"stderr"},
+		ErrorOutputPaths: []string{"stderr"},
+	}
+	return zapCfg.Build(options...)
+}
+
 // Run starts the collector according to the given configuration, and waits for it to complete.
 // Consecutive calls to Run are not allowed, Run shouldn't be called once a collector is shut down.
 func (col *Collector) Run(ctx context.Context) error {
 	if err := col.setupConfigurationComponents(ctx); err != nil {
 		col.setCollectorState(StateClosed)
+		logger, loggerErr := newFallbackLogger(col.set.LoggingOptions)
+		if loggerErr != nil {
+			fmt.Printf("unable to create fallback logger, %v", err)
+		} else {
+			if col.bc != nil {
+				for _, log := range col.bc.TakeLogs() {
+					logger.Log(log.Level, log.Message, log.Context...)
```

Comment:
Why not do the same as above? (`Check` + `Write`)

Reply:
I forgot to update this instance, but you're right, it should use the same logic.